src/GapicClientTrait.php

```diff
@@ -528,6 +528,98 @@ private static function defaultTransport()
+        }
+        $method = $this->descriptors[$methodName];
+
+        // Ensure required descriptor configuration exists.
+        if (!isset($method['callType'])) {
+            throw new ValidationException("Requested method '$methodName' does not have a callType " .
+                "in descriptor configuration.");
+        }
+        $callType = $method['callType'];
+
+        // Validate various callType specific configurations.
+        if ($callType == Call::LONGRUNNING_CALL) {
+            if (!isset($method['longRunning'])) {
+                throw new ValidationException("Requested method '$methodName' does not have a longRunning config " .
+                    "in descriptor configuration.");
+            }
+            // @TODO: check if the client implements `OperationsClientInterface` instead
+            if (!method_exists($this, 'getOperationsClient')) {
+                throw new ValidationException("Client missing required getOperationsClient " .
+                    "for longrunning call '$methodName'");
+            }
+        } elseif ($callType == Call::PAGINATED_CALL) {
+            if (!isset($method['pageStreaming'])) {
+                throw new ValidationException("Requested method '$methodName' with callType PAGINATED_CALL does not " .
+                    "have a pageStreaming in descriptor configuration.");
+            }
+        }
+        
+        // LRO are either Standard LRO response type or custom, which are handled by
+        // startOperationCall, so no need to validate responseType for those callType.
+        if ($callType != Call::LONGRUNNING_CALL) {
+            if (!isset($method['responseType'])) {
+                throw new ValidationException("Requested method '$methodName' does not have a responseType " .
+                    "in descriptor configuration.");
+            }
+        }
+    }
+
+    /**
+     * @param string $methodName
+     * @param Message $request
+     * @param array $optionalArgs {
+     *     Call Options
+     *
+     *     @type array $headers [optional] key-value array containing headers
+     *     @type int $timeoutMillis [optional] the timeout in milliseconds for the call
+     *     @type array $transportOptions [optional] transport-specific call options
+     *     @type RetrySettings|array $retrySettings [optional] A retry settings
+     *           override for the call.
+     * }
+     *
+     * @experimental
+     *
+     * @return PromiseInterface
+     */
+    private function startAsyncCall(
+        string $methodName,
+        Message $request,
+        array $optionalArgs = []
+    ) {
+        // Convert method name to the UpperCamelCase of RPC names from lowerCamelCase of GAPIC method names
+        // in order to find the method in the descriptor config.
+        $methodName = ucfirst($methodName);
+        $this->validateCallConfig($methodName);
+        
+        $method = $this->descriptors[$methodName];
```

Comment:
Having the `validateCallConfig` method return the method descriptors would clean this up!

Also, nit, but make variable name `$methodDescriptors` for clarity?
```suggestion
        $methodDescriptors = $this->validateCallConfig($methodName);
```

Reply:
For sure, sounds good! I will implement those changes.

Reply:
Done.